src/Tizen.NUI.Gadget/Tizen.NUI/NUIGadgetManager.cs

```diff
@@ -0,0 +1,294 @@
+        }
+
+        private static void OnLowBattery(object sender, LowBatteryEventArgs args)
+        {
+            HandleEvents(NUIGadgetEventType.LowBattery, args);
+        }
+
+        private static void OnLocaleChanged(object sender, LocaleChangedEventArgs args)
+        {
+            HandleEvents(NUIGadgetEventType.LocaleChanged, args);
+        }
+
+        private static void OnRegionFormatChanged(object sender, RegionFormatChangedEventArgs args)
+        {
+            HandleEvents(NUIGadgetEventType.RegionFormatChanged, args);
+        }
+
+        private static void OnDeviceOrientationChanged(object sender, DeviceOrientationEventArgs args)
+        {
+            HandleEvents(NUIGadgetEventType.DeviceORientationChanged, args);
+        }
+
+        /// <summary>
+        /// Occurs when the lifecycle of the NUIGadget is changed.
+        /// </summary>
+        /// <since_tizen> 11 </since_tizen>
+        public static event EventHandler<NUIGadgetLifecycleChangedEventArgs> NUIGadgetLifecycleChanged;
+
+        private static void OnNUIGadgetLifecycleChanged(object sender, NUIGadgetLifecycleChangedEventArgs args)
+        {
+            NUIGadgetLifecycleChanged?.Invoke(sender, args);
+
+            if (args.State == NUIGadgetLifecycleState.Destroyed)
+            {
+                args.Gadget.LifecycleChanged -= OnNUIGadgetLifecycleChanged;
+                _gadgets.Remove(args.Gadget);
+            }
+        }
+
+        /// <summary>
+        /// Adds a NUIGadget to a NUIGadgetManager.
+        /// </summary>
+        /// <param name="resourceType">The resource type of the NUIGadget package.</param>
+        /// <param name="className">The class name of the NUIGadget.</param>
+        /// <returns>The NUIGadget object.</returns>
+        /// <exception cref="ArgumentException">Thrown when failed because of a invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of an invalid operation.</exception>
+        /// <since_tizen> 11 </since_tizen>
+        public static NUIGadget Add(string resourceType, string className)
+        {
+            if (!_gadgetInfos.TryGetValue(resourceType, out NUIGadgetInfo info))
+            {
+                throw new ArgumentException("Failed to find NUIGadgetInfo. resource type: " + resourceType);
+            }
+
+            Assembly assembly = null;
+            try
+            {
+                Log.Warn("NUIGadgetAssembly.Load(): " + info.ResourcePath + info.ExecutableFile + " ++");
+                assembly = Assembly.Load(File.ReadAllBytes(info.ResourcePath + info.ExecutableFile));
+                Log.Warn("NUIGadgetAssembly.Load(): " + info.ResourcePath + info.ExecutableFile + " --");
+            }
+            catch (FileLoadException e)
+            {
+                throw new InvalidOperationException(e.Message);
+            }
+
+            NUIGadget gadget = (NUIGadget)assembly.CreateInstance(className);
```

Comment:
If the developer accidentally does not inherit NUIGadget type, it may not be possible to change to the type.
How about using 'as' and check the result?
```cs
NUIGadget gadget = assembly.CreateInstance(className) as NUIGadget;
if (gadget == null)
    throw new InvalidOperationException(...); 
```

Reply:
Done. Thanks.